dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java

```diff
@@ -499,4 +501,27 @@ protected String joinValidInvokerAddresses() {
             .map(URL::getAddress)
             .collect(Collectors.joining(","));
     }
+
+    private Map<ServiceType, Map<String, Integer>> getSummary() {
+        Map<ServiceType, Map<String, Integer>> summaryMap = new HashMap<>();
+
+        summaryMap.put(ServiceType.D_VALID, groupByServiceKey(getValidInvokers()));
+        summaryMap.put(ServiceType.D_DISABLE, groupByServiceKey(getDisabledInvokers()));
+        summaryMap.put(ServiceType.D_TO_RECONNECT, groupByServiceKey(getInvokersToReconnect()));
+        summaryMap.put(ServiceType.D_ALL, groupByServiceKey(getInvokers()));
+        return summaryMap;
+    }
+
+    private Map<String, Integer> groupByServiceKey(Collection<Invoker<T>> invokers) {
+
+        Map<String, Integer> serviceNumMap = new HashMap<>();
+        if (invokers.size() > 0) {
+            serviceNumMap = invokers.stream().filter(invoker -> invoker.getInterface() != null).collect(Collectors.groupingBy(invoker -> invoker.getInterface().getName(), Collectors.reducing(0, e -> 1, Integer::sum)));
+            if (invokers.stream().anyMatch(invoker -> invoker.getInterface() == null)) {
+                serviceNumMap.clear();
```

Comment:
Why clear here

Reply:
![image](https://user-images.githubusercontent.com/38374721/230382382-1ccd960c-16a1-4cba-a947-22d9de8a3e0c.png)
An error will be reported without this line